datafusion/core/tests/sqllogictests/test_files/predicates.slt

```diff
@@ -249,6 +249,37 @@ SELECT * FROM test WHERE column1 IN ('foo', 'Bar', 'fazzz')
 foo
 fazzz
 
+
+# Test logical plan simplifies large OR chains
```

Comment:
We can add more case about `large or` to cover up other shape expression tree.
like 
```
   /     \
 /  \    /  \
e1 e2    e3 e4
```
current expression tree all is `left deep tree`

Reply:
Yes, I'll add more test cases